java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -114,8 +163,20 @@ class DatabaseImpl(
     @VisibleForTesting
     suspend fun insertOrUpdate(storageKey: StorageKey, entity: Entity) =
         writableDatabase.useTransaction {
+            // Fetch/create the entity's type ID.
             val schemaTypeId = getSchemaTypeId(entity.schema)
+            // Set the type ID for this storage key.
             val storageKeyId = getStorageKeyId(storageKey)
+            insertWithOnConflict(
+                TABLE_ENTITIES,
+                null,
+                ContentValues().apply {
+                    put("storage_key_id", storageKeyId)
+                    put("type_id", schemaTypeId)
```

Comment:
Since the table name is a constant, maybe the column names should be too?

Reply:
:-/ At a certain point, it all gets a bit too confusing having everything behind constants IMO. I'd honestly prefer keeping everything inlined rather than moving everything to constants.  

Once I've finished the impl, I can come back and make constants for everything in one fell swoop, if you'd like.